docs/source/trainer_cli.rst

```diff
@@ -0,0 +1,156 @@
+Trainer CLI and config files
+----------------------------
+
+Another source of boilerplate code that Lightning can help to reduce is in the
+implementation of training command line tools. Furthermore, it provides a
+standardized way to configure trainings using a single file that includes
+settings for :class:`~pytorch_lightning.trainer.trainer.Trainer` and user
+extended :class:`~pytorch_lightning.core.lightning.LightningModule` and
+:class:`~pytorch_lightning.core.datamodule.LightningDataModule` classes. The
+full configuration is automatically saved in the log directory. This has the
+benefit of greatly simplifying the reproducibility of experiments.
+
+The main requirement for user extended classes to be made configurable is that
+all relevant init arguments must have type hints. This is not a very demanding
+requirement since anyway it is good practice to do anyway. As a bonus if the
+arguments are also described in the docstrings, then the help of the training
+tool will display them.
+
+----------
+
+
+training_cli
+^^^^^^^^^^^^
+
+The case in which the user's :class:`LightningModule` class implements all
+required :code:`*_dataloader` methods, a :code:`trainer.py` tool can be as
+simple as:
+
+.. code-block:: python
+
+    from pytorch_lightning.utilities.jsonargparse_utils import trainer_cli
+    from mycode import LitModel
+
+    trainer_cli(LitModel)
+
+The help of the tool describing all configurable options and default values can
+be shown by running :code:`python trainer.py --help`. Default options can be
+changed by providing individual command line arguments. However, it is better
+practice to create a configuration file and provide this to the trainer. A way
+to do this would be:
+
+.. code-block:: bash
+
+    # Dump default configuration to have as reference
+    python trainer.py --print-config > default_config.yaml
```

Comment:
Lightning uses underscore to split argument names (e.g. `--print_config`). I see this argument is defined inside jsonargparse. Can you add an alias for consistency?

Reply:
Will do

Reply:
Done